### DISCUSSION TITLE
Limitation of F' on larger scale spacecraft

### DISCUSSION THREAD

**Discussion by rawinza555** [nasa/fprime#1020] [Q&A] [answered]
It's been a while since I started picking up F' and I'm loving it so far. However, on numerous occasion, I have seen some mentions about F' suitable for small scale satellite / CubeSat. I'm wondering what  would make F' not an ideal flight control software framework for larger scale spacecraft, let's say a constellation of small satellites (Like StarLink), Heavy satellite (Like GPS satellite) or manned space capsule (Like Crew Dragon Capsule)? 

In other words, is there any technical aspect of the framework that affect its scalability?

My guess is on the additional complexity of the topology or something about the need of more CPU cores but I don't think that's it though.

**Comment by Joshua-Anderson** [comment, marked as answer]
There's no real technical limitations for why F' couldn't be used on larger missions. Larger missions often have much higher verification and validation requirements, for example 100% unit test code coverage, that F' may not meet yet.

**Reply by timcanham** [reply]
Agreed - it can be scaled up to larger projects. The topology can be more challenging to manage when there are larger topologies. We are working towards making those easier with better modeling.

**Reply by rawinza555** [reply]
Make sense! Thank you for the input.
